Sources/Connection.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  Connection.swift
+//  C7
+//
+//  Created by Scott Byrns on 4/1/16.
+//
+//
+
+public protocol Connection {
+    
+    associatedtype StatusType
+    
+    func open() throws
+    func close() -> Bool
+    
+    var connectionInfo: ConnectionInfo { get }
+    var status: StatusType { get }
+    var mostRecentError: ErrorProtocol? { get }
```

Comment:
what's the status? I'm not a fan of `associatedtype`s in `SwiftX` cause they constraint the type to `Self` and this is terrible for compatibility.